examples/src/ColorPicker.tsx

```diff
@@ -54,14 +51,28 @@ const ColorPicker = ({
             if (selectedSubagent === "<unmodified>") {
                 subAgent = [""];
             }
-            const entry: SelectionEntry = {
-                name: selectedAgent,
-                tags: subAgent,
-            };
-            setColorSelectionInfo({
-                agent: entry,
-                color: selectedColor,
+            const appliedColors = uiDisplayData.map((agent) => {
+                const newAgent = { ...agent };
+                if (agent.name === selectedAgent) {
+                    if (subAgent.includes("")) {
+                        newAgent.color = selectedColor;
+                    }
+                    const newDisplayStates = agent.displayStates.map(
+                        (state: any) => {
+                            if (subAgent.includes(state.id)) {
+                                return {
+                                    ...state,
+                                    color: selectedColor,
+                                };
+                            }
+                            return state;
+                        }
+                    );
+                    newAgent.displayStates = newDisplayStates;
+                }
+                return newAgent;
```

Comment:
Instead of passing in a `colorChange` we send in a `uiDisplayData`

In pairing w @meganrm we decided to make the front end responsible for generating and storing this state, viewer renders